src/components/tables/TableRowReducers.ts

```diff
@@ -6,26 +6,30 @@ import * as _ from 'underscore';
 export interface ITableRowState {
   id: string;
   opened: boolean;
+  selected: boolean;
 }
 
-export const tableRowInitialState: ITableRowState = { id: undefined, opened: undefined };
+export const tableRowInitialState: ITableRowState = { id: undefined, opened: undefined, selected: undefined };
 export const tableRowsInitialState: ITableRowState[] = [];
 
 export const tableRowReducer = (state: ITableRowState = tableRowInitialState, action: IReduxAction<IReduxActionsPayload>): ITableRowState => {
   switch (action.type) {
     case TableRowActions.add:
       return {
         id: action.payload.id,
-        opened: false
+        opened: false,
+        selected: false,
       };
     case TableRowActions.toggle:
```

Comment:
Should the action be named `TableRowActions.select` ? It makes more sense IMHO since you're not always toggling it but always changing the `selected` attribute